airflow/models/renderedtifields.py

```diff
@@ -0,0 +1,153 @@
+from airflow.utils.session import provide_session
+from airflow.utils.sqlalchemy import UtcDateTime
+
+
+class RenderedTaskInstanceFields(Base):
+    """
+    Save Rendered Template Fields
+    """
+
+    __tablename__ = "rendered_task_instance_fields"
+
+    dag_id = Column(String(ID_LEN), primary_key=True)
+    task_id = Column(String(ID_LEN), primary_key=True)
+    execution_date = Column(UtcDateTime, primary_key=True)
+    rendered_fields = Column(JSON, nullable=False)
+
+    def __init__(self, ti: TaskInstance, render_templates=True):
+        self.dag_id = ti.dag_id
+        self.task_id = ti.task_id
+        self.task = ti.task
+        self.execution_date = ti.execution_date
+        self.ti = ti
+        if render_templates:
+            ti.render_templates()
+        self.rendered_fields = {
+            field: serialize_template_field(
+                getattr(self.task, field)
+            ) for field in self.task.template_fields
+        }
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}: {self.dag_id}.{self.task_id} {self.execution_date}"
+
+    @classmethod
+    @provide_session
+    def get_templated_fields(cls, ti: TaskInstance, session: Session = None) -> Optional[dict]:
+        """
+        Get templated field for a TaskInstance from the RenderedTaskInstanceFields
+        table.
+
+        :param ti: Task Instance
+        :param session: SqlAlchemy Session
+        :return: Rendered Templated TI field
+        """
+        result = session.query(cls.rendered_fields).filter(
+            cls.dag_id == ti.dag_id,
+            cls.task_id == ti.task_id,
+            cls.execution_date == ti.execution_date
+        ).first()
+
+        if result:
+            rendered_fields = result.rendered_fields
+            if isinstance(rendered_fields, str):
+                rendered_fields = json.loads(rendered_fields)
+            return rendered_fields
+        else:
+            return None
+
+    @classmethod
+    @provide_session
+    def has_templated_fields(cls, ti: TaskInstance, session: Session = None) -> bool:
+        """Checks templated field exist for this ti.
+
+        :param ti: Task Instance
+        :param session: SqlAlchemy Session
+        """
+        # We cant use q.exists() because some databases such as SQL Server don’t allow
+        # an EXISTS expression to be present in the columns clause of a SELECT.
+        # Details: https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.exists
+        return session.query(exists().where(
+            and_(cls.dag_id == ti.dag_id,
+                 cls.task_id == ti.task_id,
+                 cls.execution_date == ti.execution_date)
+        )).scalar()
+
+    @provide_session
+    def write(self, session: Session = None):
+        """Write instance to database
+
+        :param session: SqlAlchemy Session
+        """
+        session.merge(self)
```

Comment:
Do we need this method? This only makes it difficult to understand the code, because it is necessary to look at the content of the method to see that it invokes a database query.

Reply:
It is a separate method as in the future we might want to add some validation or might want to modify the parameters. 

Also, we don't "commit" inside this function and let the caller decide when to save this object to Database when re-using the session.